src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -14,14 +16,15 @@ object UserService {
     def pure[A](a: A): DBIO[A] = DBIO.successful(a)
     def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
   }
-
+  
   def createUser(createdUser: User)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], UserId]] = {
     val resT = for {
       validUser <- XorT[DBIO, NonEmptyList[ValidationError], User](User.validateUser(createdUser.email, createdUser.password))
-      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(validUser))
+      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(hashUserPassword(validUser)))
     } yield userId
 
     resT.value
   }
 
+  private def hashUserPassword(user: User): User = user.copy(password = user.password.bcrypt)
```

Comment:
Well it's now hashed in the monad which describes the process/computation of creating and saving in the DB an User. After the password has been validated using validateUser and just before saving in the DB it gets hashed.
The other option is to use there NewUser and making User hash the passed password the moment it's created. This is pretty logical as User represents the DB table and in the DB the password should always be hashed.